sirepo/template/genesis.py

```diff
@@ -0,0 +1,247 @@
+    'yrms',
+    'error',
+)
+
+_LATTICE_DATA_FILENAME = 'lattice.npy'
+
+_LATTICE_RE = re.compile(r'^.+power[\s\w]+\n(.*)', flags=re.DOTALL)
+
+# POSIT: Same name as outputfile in schema
+_OUTPUT_FILENAME = 'genesis.out'
+_FIELD_DISTRIBUTION_OUTPUT_FILENAME = _OUTPUT_FILENAME + '.fld'
+_PARTICLE_OUTPUT_FILENAME = _OUTPUT_FILENAME + '.par'
+
+_RUN_ERROR_RE = re.compile(r'(?:^\*\*\* )(.*error.*$)', flags=re.MULTILINE)
+
+# POSIT: Same order as results in _OUTPUT_FILENAME
+_SLICE_COLS = (
+    'z[m]',
+    'aw',
+    'qfld',
+)
+
+_SLICE_DATA_FILENAME = 'slice.npy'
+
+_SLICE_RE = re.compile(
+    r'\s+z\[m\]\s+aw\s+qfld\s+\n(.*)\n^\s*$\n\*',
+    flags=re.DOTALL|re.MULTILINE,
+)
+
+def background_percent_complete(report, run_dir, is_running):
+    if is_running:
+        return PKDict(percentComplete=0, frameCount=0)
+    if not _genesis_success_exit(run_dir):
+        return PKDict(
+            percentComplete=100,
+            state=sirepo.job.ERROR,
+        )
+    return PKDict(
+        percentComplete=100,
+        frameCount=_get_field_distribution(
+            sirepo.simulation_db.read_json(
+                run_dir.join(template_common.INPUT_BASE_NAME),
+            ),
+        ).shape[0],
+    )
+
+
+def post_execution_processing(run_dir=None, **kwargs):
+    if _genesis_success_exit(run_dir):
+        return
+    return _parse_genesis_error(run_dir)
+
+
+def sim_frame_fieldDistributionAnimation(frame_args):
+    r = _get_field_distribution(frame_args.sim_in)
+    d = np.abs(r[int(frame_args.frameIndex), 0, :, :])
+    s = d.shape[0]
+    return PKDict(
+        title='Field Distribution',
+        x_label='',
+        x_range=[0, s, s],
+        y_label='',
+        y_range=[0, s, s],
+        z_matrix=d.tolist(),
+    )
+
+
+def sim_frame_parameterAnimation(frame_args):
+    l, s = _get_lattice_and_slice_data(frame_args.run_dir)
+    x = 'z[m]'
+    y = frame_args.y
+    return template_common.parameter_plot(
+        s[:, _SLICE_COLS.index(x)].tolist(),
+        [PKDict(
+            field=y,
+            points=l[:, _LATTICE_COLS.index(y)].tolist(),
+            label=y,
+        )],
+        PKDict(),
+        PKDict(
+            title=f'Slice',
+            x_label=x,
+            y_label=y,
+        )
+    )
+
+
+def sim_frame_particleAnimation(frame_args):
+    def _get_col(col_key):
+        # POSIT: ParticleColumn keys are in same order as columns in output
+        for i, c in enumerate(_SCHEMA.enum.ParticleColumn):
+            if c[0] == col_key:
+              return i, c[1]
+        raise AssertionError(
+            f'No column={_SCHEMA.enum.ParticleColumn} with key={col_key}',
+        )
+    n = frame_args.sim_in.models.electronBeam.npart
+    d = np.fromfile(_PARTICLE_OUTPUT_FILENAME , dtype=np.float64)
+    b = d.reshape(
+        int(len(d) / len(_SCHEMA.enum.ParticleColumn) / n),
+        len(_SCHEMA.enum.ParticleColumn),
+        n,
+    )
+    x = _get_col(frame_args.x)
+    y = _get_col(frame_args.y)
```

Comment:
All other sirepo particle plots have been heatmaps, which shows how dense the distribution is. Is there a reason we want to show Genesis particles as points?

Reply:
Thanks @moellep . Stephen agreed that a heatmap makes more sense. https://github.com/radiasoft/sirepo/pull/3831/commits/7943007990b2b270e7df303130a1e1df8438e621